### PR TITLE
Add a new DAG to upload AMO JSON for TAAR

### PR DESCRIPTION
**Pull request by crankycoder**
This adds a new DAG responsible for uploading a JSON file of AMO data for consumption by TAAR.

Also see: https://github.com/mozilla/python_mozetl/pull/207

### CODE REVIEW

File: dags/taar_amodump.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    'depends_on_past': False,
+    'start_date': datetime(2018, 3, 20),
+    'email': ['telemetry-alerts@mozilla.com', 'mreid@mozilla.com'],
+    'email_on_failure': True,
+    'email_on_retry': True,
+    'retries': 3,
+    'retry_delay': timedelta(minutes=30),
+}
+
+try:
```
Comment (acmiyaguchi) [inline dags/taar_amodump.py]: The try block is unnecessary here.

File: dags/taar_amodump.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        task_id="taar_amodump",
+        job_name="Dump AMO JSON blobs with oldest creation date",
+        execution_timeout=timedelta(hours=1),
+        instance_count=1,
+        owner="vng@mozilla.com",
+        email=["mlopatka@mozilla.com", "vng@mozilla.com", "sbird@mozilla.com"],
+        env=mozetl_envvar("taar_amodump", {
+            "path": "/tmp/amo_cache",
+            "date": "{{ ds_nodash }}"
+        }),
```
Comment (acmiyaguchi) [inline dags/taar_amodump.py]: You'll need to pass the extra argument to specify the `MOZETL_SUBMISSION_METHOD`.  `mozetl_envvar` takes an "other" dictionary that you can pass arbitrary environment variables through. 

```python
moz_envvar("taar_amodump", { ... }, {"MOZETL_SUBMISSION_METHOD": "python"})
```

Comment (crankycoder) [reply dags/taar_amodump.py]: crap. thanks for catching this.

File: dags/taar_amodump.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+amodump = EMRSparkOperator(
+    task_id="taar_amodump",
+    job_name="Dump AMO JSON blobs with oldest creation date",
+    execution_timeout=timedelta(hours=1),
+    instance_count=1,
+    owner="vng@mozilla.com",
+    email=["mlopatka@mozilla.com", "vng@mozilla.com", "sbird@mozilla.com"],
+    env=mozetl_envvar("taar_amodump",
+                      {"path": "/tmp/amo_cache",
+                       "date": "{{ ds_nodash }}"},
```
Comment (acmiyaguchi) [inline dags/taar_amodump.py]: Specifying the bucket as  `{{ task.__class__.private_output_bucket }}` will make testing easier. On production, this is `telemetry-parquet`. On dev, its `telemetry-test-bucket`.

**Comment by crankycoder** [comment]
My docker container just finished up and I've got a new extended_addons_database.json file up on s3.

```
$ aws s3 ls s3://telemetry-parquet/telemetry-ml/addon_recommender/extended_addons_database.json
2018-04-03 15:20:25   56986065 extended_addons_database.json
```

**Comment by acmiyaguchi** [comment]
Looks good to me, thanks for the hard work.

**Pull request merged**